sdk/demos/intel/js/renderer.js

```diff
@@ -0,0 +1,236 @@
+    this.gl = gl;
+
+    this.texture = gl.createTexture();
+    this.program = gl.createProgram();
+
+    this.enable_WEBGL_video_texture = enable_WEBGL_video_texture;
+
+    // Create a vertex shader object
+    var verts = [
+      -1.0,  1.0, 0.0, 0.0, 0.0,// Top-left
+      -1.0, -1.0, 0.0, 0.0, 1.0,// Bottom-left
+      1.0, -1.0, 0.0, 1.0, 1.0,// Bottom-right
+      1.0,  1.0, 0.0, 1.0, 0.0,// Top-right
+    ];
+
+    this.indices = [3, 2, 1, 3, 1, 0]
+
+    this.vertBuffer = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, this.vertBuffer);
+    gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(verts), gl.STATIC_DRAW);
+    gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+    this.indexBuffer = gl.createBuffer();
+    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, this.indexBuffer);
+    gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array(this.indices), gl.STATIC_DRAW);
+    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, null);
+
+    // Create program object
+    var vertShader = gl.createShader(gl.VERTEX_SHADER);
+    gl.shaderSource(vertShader, VS);
+    gl.compileShader(vertShader);
+
+    var fragShader = gl.createShader(gl.FRAGMENT_SHADER);
+    if (this.enable_WEBGL_video_texture ) {
+      this.ext = gl.getExtension('WEBGL_video_texture');
+      if (!this.ext) {
+        console.log("WEBGL_video_texture not supported");
+        return;
+      }
+      gl.shaderSource(fragShader, WEBGLVideoTextureFS);
+    } else {
+      gl.shaderSource(fragShader, FS);
+    }
+    gl.compileShader(fragShader);
+
+    gl.attachShader(this.program, vertShader);
+    gl.attachShader(this.program, fragShader);
+    gl.linkProgram(this.program);
+
+    this.videoElement = null;
+  };
+
+  Renderer.prototype.setVideo = function (url) {
+    var gl = this.gl;
+    var self = this;
+    var using_WEBGL_video_texture = this.enable_WEBGL_video_texture;
+    var ext = this.ext;
+
+    return new Promise(function(resolve, reject) {
+      var video = document.createElement('video');
+
+      video.addEventListener('canplay', function() {
+        // Added "click to play" UI?
+      });
+
+      video.addEventListener('playing', function() {
+        self.videoElement = video;
+        self.imgElement = null;
+
+        if (!using_WEBGL_video_texture) {
+          gl.bindTexture(gl.TEXTURE_2D, self.texture);
+          gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, self.videoElement);
+
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+          gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+        } else {
+          if (this.ext !== null) {
+            gl.bindTexture(ext.TEXTURE_VIDEO_IMAGE, self.texture);
+            ext.VideoElementTargetVideoTexture(ext.TEXTURE_VIDEO_IMAGE, video);
```

Comment:
Should this be using shareVideoImageWEBGL?

Reply:
I'm sorry. In a long time ago, I think there is some disagree with changing the name to shareVideoImageWEBGL and without a proper name. So I remember the decision should be keep the VideoElementTargetVideoTexture. (I'll update the spec if this is true)

Reply:
I vaguely remember there was some disagreement about the share/release semantics too, but I can't find it in the WebGL working group's meeting minutes folder. If you can search in there and you find it, please tell us which document you found it in.

At this point, https://www.khronos.org/registry/webgl/extensions/proposals/WEBGL_video_texture/ defines the share/release APIs. The share/release semantic is absolutely necessary for some implementations - like Chrome's - to minimize the duration the video frames are locked. So, could you please update Blink's IDL and implementation to follow the current extension proposal, and then update this demo to use the share/release APIs? Thanks.